### ISSUE TITLE
Index Constraints owned by a PostgreSQL group role, not a user role

### ISSUE THREAD

**Issue by bradyclifford** [JasperFx/marten#796]
In our Postgres environment, every object is owned by a group role, not a user role.  This allows multiple user roles to modify schema when needed. We don't give permissions to a user role directly because it's not the same from one environment to the next.

In Marten when doing the schema check, the query it uses for getting all of the indices requires that the index is owned by a User role.
```JOIN pg_user AS U ON i.relowner = U.usesysid```

Marten is not able to find our new index constraint and it will throw an exception when we have auto create disabled in our environments.

**Comment by jeremydmiller** [comment]
@bradyclifford Can you just get around this by going AutoCreate.None in other environments? At this point, that never checks the schema itself. I'm hesitant to do any more customization of Marten to acomodate our database team.

**Comment by bradyclifford** [comment]
AutoCreate.None doesn't fix the problem to my recollection.  It has been awhile, so I will have to triage it again.

Andrew might have additional details.

**Comment by bradyclifford** [comment]
To summarize, there are 2 issues.
1. `AutoCreate.None` is still asserting schema
1. `pg_roles` should be used instead of `pg_user` when asserting; it will still include any LOGIN user roles

As an example:
```
SELECT
  R.rolname                AS role_name,
  ns.nspname               AS schema_name,
  pg_catalog.textin(pg_catalog.regclassout(idx.indrelid :: REGCLASS)) AS table_name,
  i.relname                AS index_name,
  pg_get_indexdef(i.oid) as ddl,
  idx.indisunique          AS is_unique,
  idx.indisprimary         AS is_primary,
  am.amname                AS index_type,
  idx.indkey,
       ARRAY(
           SELECT pg_get_indexdef(idx.indexrelid, k + 1, TRUE)
           FROM
             generate_subscripts(idx.indkey, 1) AS k
           ORDER BY k
       ) AS index_keys,
  (idx.indexprs IS NOT NULL) OR (idx.indkey::int[] @> array[0]) AS is_functional,
  idx.indpred IS NOT NULL AS is_partial
FROM pg_index AS idx
  JOIN pg_class AS i
    ON i.oid = idx.indexrelid
  JOIN pg_am AS am
    ON i.relam = am.oid
  JOIN pg_namespace AS NS ON i.relnamespace = NS.OID
  JOIN pg_roles AS R ON i.relowner = U.oid
WHERE NOT nspname LIKE 'pg%' AND i.relname like 'mt_%'; -- Excluding system table
```

**Comment by bradyclifford** [comment]
We finally have been able to convince our DB team to allow us to have `AutoCreate.CreateOrUpdate` in production turned on.

However Marten complains when this is on that the constraints don't exist, when they do explained above.

Can this issue be revisited and a fix applied to `master` but also tag `v2.8.0`?
